### PR TITLE
bbcbc: Input overhaul and machine/software list corrections

### PR DESCRIPTION
**Pull request by ajrhacker**
- Button mappings completely reworked: grey bidding/suit selection buttons classified as keypad inputs and put in correct order, and other buttons as console-standard Start, Select, Button 1-2
- Inputs now read through PIO instead of RAM addresses, making them less hypersensitive
- Manufacturer/publisher is Unicard, as manuals and system/cartridge labels clearly state
- Added release years for some common cartridges
- (nw) Link bbcbc in its own library rather than the unrelated libacorn ("make clean" strongly recommended)

### CODE REVIEW

**Comment by cuavas** [comment]
@Pernod70 as our resident expert on all things with BBC in the name, any objections to this?  The code looks reasonable.

I'm not sure about about making a library just for this machine though.  Maybe it should be moved to a misc library rather than getting its own?

**Comment by ajrhacker** [comment]
The single-driver library is following the MESS convention for non-arcade machines, where only skeleton drivers are grouped together without regard to manufacturer. The BBC Bridge Companion and BBC Micro have basically no relation to each other aside from using the BBC name under license.

**Comment by Pernod70** [comment]
No objections here, I know nothing about this system but glad it's removed from acorn library as it is completely unrelated.

**Comment by startaq** [comment]
Shouldn't it be `libunicard.a` then?

File: src/mame/drivers/bbcbc.cpp [hunk 1]
```diff
@@ -111,40 +120,63 @@ static const z80_daisy_config bbcbc_daisy_chain[] =
...
+			return ioport("BUTTONS1")->read();
+		case 0xdf:
+			return ioport("BUTTONS2")->read();
+		case 0xbf:
+			return ioport("BUTTONS3")->read();
+	}
+	logerror("Unknown input select: %02x\n", m_input_select);
+	return 0xff;
+}
+
```
Comment (startaq) [inline src/mame/drivers/bbcbc.cpp]: Why not write it like this (probably closer to hardware):

`UINT8 data = 0xff;`
`if (BIT(m_input_select, 4) == 0) data &= ioport("BUTTONS1")->read();`
`if (BIT(m_input_select, 5) == 0) data &= ioport("BUTTONS2")->read();`
`if (BIT(m_input_select, 6) == 0) data &= ioport("BUTTONS3")->read();`
`return data;`

**Comment by rb6502** [comment]
I'm more concerned with the every frame string lookup on the ioport name.  On PCs, it's nothing for a system like this, but on ARM targets every bit helps.

**Pull request merged**